.circleci/config.yml

```diff
@@ -11,11 +15,62 @@ executors:
     docker:
     - image: circleci/golang:1.13
 
+commands:
+  publish_image:
+    parameters:
+      registry:
+        description: Registry address
+        type: string
+        default: "docker.io"
+      organization:
+        description: Registry organization
+        type: string
+        default: prombench
+      image_name:
+        description: Name of the container image
+        type: string
+      login:
+        description: Environment variable holding the registry login
+        type: env_var_name
+        default: DOCKER_USER
+      password:
+        description: Environment variable holding the registry password
+        type: env_var_name
+        default: DOCKER_PASSWORD
+      dockerfile_path:
+        description: Path to Dockerfile
+        type: string
+        default: ""
+      context_path:
+        description: Path of docker build context
+        type: string
+        default: "."
+    steps:
+    - run:
+        name: Setup environment variables
+        command: |
+          echo "export TAG=`date +%Y%m%d`" >> $BASH_ENV
+          echo "DOCKER_REPO=<< parameters.registry >>/${DOCKER_ORG:-<< parameters.organization >>}" >> $BASH_ENV
+          echo "DOCKER_IMAGE=<< parameters.image_name >>" >> $BASH_ENV
+    - when:
+        condition: <<parameters.dockerfile_path>>
```

Comment:
I don't see that `dockerfile_path` isn't always given so I wouldn't have the condition.